### PR TITLE
Support passing arguments to the command executed by the memory analyzer

### PR DESCRIPTION
**Pull request by danpoe**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/memory-analyzer/gdb_api.cpp [hunk 1]
```diff
@@ -107,15 +107,28 @@ void gdb_apit::create_gdb_process()
...
-                         const_cast<char *>(binary),
-                         NULL};
+    std::vector<std::string> exec_cmd;
+    exec_cmd.reserve(args.size() + 3);
+    exec_cmd.push_back("gdb");
+    exec_cmd.push_back("--interpreter=mi");
+    exec_cmd.push_back("--args");
+    exec_cmd.insert(exec_cmd.end(), args.begin(), args.end());
+
+    char **exec_cmd_ptr = (char **)malloc(
```
Comment (xbauch) [inline src/memory-analyzer/gdb_api.cpp]: Now that we build these using `malloc`, should we free the memory at some point?

Comment (danpoe) [reply src/memory-analyzer/gdb_api.cpp]: Yes I think we could free it in the parent process.

File: src/memory-analyzer/analyze_symbol.h [hunk 2]
```diff
@@ -34,7 +34,9 @@ class source_locationt;
 class gdb_value_extractort
 {
 public:
-  gdb_value_extractort(const symbol_tablet &symbol_table, const char *binary);
+  gdb_value_extractort(
+    const symbol_tablet &symbol_table,
+	const std::vector<std::string> &args);
```
Comment (peterschrammel) [inline src/memory-analyzer/analyze_symbol.h]: indentation

**Review by peterschrammel — COMMENTED** [review]
cpplint is complaining

File: src/memory-analyzer/gdb_api.cpp (hunk 1 above)
Comment (danpoe) [reply src/memory-analyzer/gdb_api.cpp]: Ah I just realised we're allocating memory after the `fork()` call in the child process only. And that memory is freed when the process image is replaced via `execvp()`.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: c6f9e44c2f926efc850b8d0f891c1b0b2db67334).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/118021176

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**